extras/src/main/java/com/google/gson/typeadapters/UtcDateTypeAdapter.java

```diff
@@ -46,16 +43,14 @@ public void write(JsonWriter out, Date date) throws IOException {
   @Override
   public Date read(JsonReader in) throws IOException {
     try {
-      switch (in.peek()) {
-      case NULL:
-        in.nextNull();
-        return null;
-      default:
+        if (Objects.requireNonNull(in.peek()) == JsonToken.NULL) {
```

Comment:
`Objects.requireNonNull` seems to be redundant; if `in.peek()` returns `null`, then there is an implementation bug in `JsonReader`

Reply:
I think this particular change isn't a clear improvement so I'd be inclined to keep the original `switch`.